### ISSUE TITLE
[invariant_booleans] List.length related false positive

### ISSUE THREAD

**Issue by modulovalue** [dart-lang/linter#2473]
```dart
void abc(List<int> input) {
  if (input.length >= 2) {
    input.removeLast();
    input.removeLast();
    if (input.length < 2) { // Conditions should not unconditionally evaluate to `true` or to `false`. verify: chars.length >= 2.
      print("Branch A");
    } else {
      print("Branch B");
    }
  }
}

void main() {
  abc([1, 2, 3]);
  abc([1, 2, 3, 4]);
}
```

Both branches are visited in main. 
Is this a false positive or is the 'verify: chars.length >= 2' supposed to be a call to action?

**Comment by srawlins** [comment]
We've deprecated `invariant_booleans` and will not be fixing correctness bugs going forward.
